RoutesApp_ios/repository/local/managers/LocalDataManager.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  LocalDataManager.swift
+//  RoutesApp_ios
+//
+//  Created by Yawar Valeriano on 10/3/22.
+//
+
+import Foundation
+
+protocol LocalDataManagerProtocol {
+    func retrieveAllDataFromFirebase(completion: @escaping(Result<Void, Error>) -> Void)
+}
+
+class LocalDataManager: LocalDataManagerProtocol {
+    static let shared = LocalDataManager()
+
+    private let tourpointsManager = TourpointsManager.shared
+    private let routelistManager = RouteListManager.shared
+    private let coreDataManager = CoreDataManager.shared
+    private let context = CoreDataManager.shared.getContext()
+
+    func retrieveAllDataFromFirebase(completion: @escaping(Result<Void, Error>) -> Void) {
+        if coreDataManager.deleteAll() {
+            tourpointsManager.getTourpointList { result in
+                switch result {
+                case.success(let tourpoints):
+                    self.getAndSaveTourpointCategoriesWithTourpoints(tourpoints: tourpoints, completion: completion)
+                case.failure(let error):
+                    completion(.failure(error))
+                }
+            }
+        } else {
+            completion(.failure(CoreDataError.Failed(cause: String.localizeString(localizedString: ConstantVariables.errorUpdatingData))))
+        }
+
+    }
+
+    private func getAndSaveTourpointCategoriesWithTourpoints(tourpoints: [Tourpoint], completion: @escaping(Result<Void, Error>) -> Void) {
```

Comment:
doesn't this already has a manager for tour points where it should be?

Reply:
I'm using the corresponding managers to get the data but the way I handle the result is different, that's why I have this logic in this manager